.github/workflows/main.yml

```diff
@@ -9,14 +9,17 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up Python 3.11
+    - name: Set up Python 3.13
       uses: actions/setup-python@v4
       with:
-        python-version: 3.11
+        python-version: 3.13
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install -r requirements.txt && pip install -r requirements_test.txt
+        pip install -r requirements.txt && pip install -r requirements_test.txt && pip install ruff
```

Comment:
Why not add it to `requirements_test.txt`?

Reply:
Hmm leaving that for when we fully switch to ruff, I hope the relevant E* rules we need will be moved out of preview soonish.